src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -326,6 +328,9 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanC
                     AddClassifiedSpansForPreviousTree(
                         classificationService, span, lastSnapshot, lastDocument, classifiedSpans);
                 }
+
+                // Ensure the syntax tree stays alive for as long as we're doing the processing.
+                GC.KeepAlive(lastSyntaxTree);
```

Comment:
❔ Why not modify `AddClassifiedSpansForCurrentTree` to take `SyntaxTree` instead of `Document`? It's the first action it's going to take anyway...

Reply:
So, we do the SyntaxTree for the C#/VB case.  But this type also works for TS/F#.  I will doc this optimization we have.

Reply:
> But this type also works for TS/F#.

Neither of these should be directly accessing this type. For both cases, we can modify the external access layer to process the syntax tree instead of the document.

Reply:
F# is used here. We can into them to return syntactic classifications through the IClassificationService api. I'm not a fan of having two apis here (one effectively just for us, and one for them). I'd prefer we all go through the same one.